access/src/main/java/zone/cogni/asquare/virtuoso/VirtuosoRdfStoreService.java

```diff
@@ -106,10 +107,11 @@ private void addData(Model model, String graphUri, boolean replace) {
       HttpResponse response = client.execute(request);
       int responseCode = response.getStatusLine().getStatusCode();
       String reason = response.getStatusLine().getReasonPhrase();
+      String responseBody = Try.of(() -> IOUtils.toString(response.getEntity().getContent(), StandardCharsets.UTF_8)).getOrElse(StringUtils.EMPTY);
       EntityUtils.consume(response.getEntity());
       if (!HttpStatus.valueOf(responseCode).is2xxSuccessful()) {
-        log.error("Virtuoso server send response with status code {}, with message {}", responseCode, reason);
-        throw new VirtuosoOperationException("Virtuoso server send response with status code " + responseCode + ", with message " + reason);
+        log.error("Virtuoso server sent response with status code {}, with message {}", responseCode, reason);
+        throw new VirtuosoOperationException("Virtuoso server sent response with status code " + responseCode + ", with message " + reason + " and body " + responseBody);
```

Comment:
put full response body in exception message, is that ok? You can add an extra field in VirtuosoOperationException if needed